contrib/src/test/scala/akka/stream/contrib/DelayFlowSpec.scala

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright (C) 2016 Lightbend Inc. <http://www.lightbend.com>
+ */
+package akka.stream.contrib
+
+import akka.stream.contrib.DelayFlow.DelayStrategy
+import akka.stream.scaladsl.Source
+import scala.language.postfixOps
+import akka.stream.testkit.scaladsl.TestSink
+import scala.concurrent.duration._
+
+class DelayFlowSpecAutoFusingOn extends { val autoFusing = true } with DelayFlowSpec
+class DelayFlowSpecAutoFusingOff extends { val autoFusing = false } with DelayFlowSpec
+
+trait DelayFlowSpec extends BaseStreamSpec {
+
+  "DelayFlow" should {
+
+    "work with empty source" in {
+      Source.empty[Int]
+        .via(DelayFlow(Duration.Zero))
+        .runWith(TestSink.probe)
+        .request(1)
+        .expectComplete()
+    }
+
+    "work with fixed delay" in {
+
+      val fixedDelay = 1 second
+
+      val elems = Vector(1, 2, 3, 4, 5, 6, 7, 8, 9, 0)
+
+      val probe = Source(elems)
+        .via(DelayFlow[Int](fixedDelay))
+        .runWith(TestSink.probe)
+        .request(elems.size)
+
+      elems.foreach(probe
+        //the timer is not precise enough and fails occasionally if exact delay value is used
+        .expectNoMsg(fixedDelay - (10 millis))
```

Comment:
I guess scheduler in TimerGraphStageLogic is not precise enough... can it be the case?